### PR TITLE
executor: exchange operator supports data compression

### PR DESCRIPTION
**Pull request by solotzg**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/40494 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### CODE REVIEW

File: proto/executor.proto [hunk 1]
```diff
@@ -61,6 +61,13 @@ enum ExchangeType {
 	Hash = 2;
 }
 
+// Data compression mode
+enum CompressionMode {
+	NONE = 0; // no compression
+	FAST = 1; // fast compression/decompression speed, compression ratio is lower than HC mode
+	HIGH_COMPRESSION = 2; // high compression (HC) ratio mode
```
Comment (windtalker) [inline proto/executor.proto:68]: Maybe we should use the name of the compression algorithm as the enum value, which is more easier if we wants to add more compression algorithms.

Comment (solotzg) [reply proto/executor.proto:68]: Originally, I use compression method name directly. But the real codec format is implemented by TiFlash itself and other components can't recognize. If we need to change compression method for one mode, we can just update [MPPDataPacket::version](https://github.com/pingcap/kvproto/blob/7cd28226c2a21e489b077a87e1f5c9ba2c950944/proto/mpp.proto#L76) and use new format.

Comment (solotzg) [reply proto/executor.proto:68]: Compression method like `snappy`, `LZ4`, `LZ4-xxx` are in same family. I think we can use mode `FAST` to present it. We can let user choose one mode by tidb session and let tiflash decide the actual method.

Comment (windtalker) [reply proto/executor.proto:68]: OK

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**